### PR TITLE
Statusing nested

### PR DESCRIPTION
**Pull request by tblackwe**
### Description
Fixes for Completion Status of nested fields

### How to test
Validate that unfilled nested fields cause submission to be blocked

### Author checklist

- [X] I have performed a self-review of my code
- [X] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [X] I have updated the documentation, if necessary

### CODE REVIEW

File: services/app-api/utils/validation/completionStatus.test.ts [hunk 1]
```diff
@@ -231,4 +236,23 @@ describe("Completion Status Tests", () => {
       });
     });
   });
+
+  describe.skip("Quick Test", () => {
```
Comment (oweingrod-coforma) [inline services/app-api/utils/validation/completionStatus.test.ts:240]: Should we still skip this?

File: services/app-api/utils/validation/completionStatus.ts [hunk 2]
```diff
@@ -66,6 +68,32 @@ export const calculateCompletionStatus = async (
...
+    ) => {
+      let selectedChoicesIds = selectedChoices
+        .map((choice: Choice) => choice.key)
+        .map((choiceId: string) => choiceId?.split("-").pop());
+      let selectedChoicesWithChildren = fieldChoices?.filter(
+        (fieldChoice: FieldChoice) =>
+          selectedChoicesIds.includes(fieldChoice.id) && fieldChoice.children
+      );
+      let fieldIds: string[] = [];
+      selectedChoicesWithChildren?.forEach((selectedChoice: any) => {
```
Comment (oweingrod-coforma) [inline services/app-api/utils/validation/completionStatus.ts:83]: Can we add non-any types to these?

File: services/app-api/utils/validation/completionStatus.test.ts (hunk 1 above)
Comment (tblackwe) [reply services/app-api/utils/validation/completionStatus.test.ts:240]: Updated name for clarity.

File: services/app-api/utils/validation/completionStatus.ts (hunk 2 above)
Comment (tblackwe) [reply services/app-api/utils/validation/completionStatus.ts:83]: ty, good catch.

**Review by oweingrod-coforma — APPROVED** [review]

**Pull request merged**